src/app/lib/presentation/dad_jokes/dad_jokes_page.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:app/business/dad_jokes/dad_joke.dart';
+import 'package:app/business/dad_jokes/dad_jokes_service.dart';
+import 'package:app/presentation/dad_jokes/dad_joke_list_item.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:get_it/get_it.dart';
+
+/// The dad jokes page.
+final class DadJokesPage extends ConsumerWidget {
+  /// Provider that provides dad jokes.
+  static final dadJokesProvider = StreamProvider<List<DadJoke>>((ref) async* {
+    final dadJokesService = GetIt.I<DadJokesService>();
+
+    await for (final dadJokes in dadJokesService.dadJokesStream) {
+      yield dadJokes;
+    }
+  });
+
+  const DadJokesPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final dadJokesAsyncValue = ref.watch(dadJokesProvider);
+    return dadJokesAsyncValue.when(
+      data: (dadJokes) {
+        return ListView.builder(
```

Comment:
Shouldn't you leverage keys so that when you set something as favorite, the whole list isn't refreshed?

Reply:
I'll investigate a little bit, I haven't used them much to be honest.

Reply:
I passed a key to my item, not sure how to validate that it works like we think though.